mobot/mobot_client/payments.py

```diff
@@ -0,0 +1,186 @@
+
+        if amount_mob <= 0:
+            return
+
+        self.send_mob_to_address(
+            source, self.account_id, amount_mob, customer_payments_address
+        )
+
+    def send_mob_to_address(
+            self, source, account_id, amount_in_mob, customer_payments_address
+    ):
+        # customer_payments_address is b64 encoded, but full service wants a b58 address
+        customer_payments_address = mc.utility.b64_public_address_to_b58_wrapper(
+            customer_payments_address
+        )
+
+        tx_proposal = self.mcc.build_transaction(
+            account_id, amount_in_mob, customer_payments_address
+        )
+        txo_id = self.submit_transaction(tx_proposal, account_id)
+        for _ in range(10):
+            try:
+                self.mcc.get_txo(txo_id)
+                break
+            except Exception:
+                print("TxOut did not land yet, id: " + txo_id)
+                pass
+            time.sleep(1.0)
+        else:
+            self.signal.send_message(
+                source,
+                "couldn't generate a receipt, please contact us if you didn't a payment!",
+            )
+            return
+
+        self.send_payment_receipt(source, tx_proposal)
+
+    def submit_transaction(self, tx_proposal, account_id):
+        # retry up to 10 times in case there's some failure with a 1 sec timeout in between each
+        transaction_log = self.mcc.submit_transaction(tx_proposal, account_id)
+        list_of_txos = transaction_log["output_txos"]
+
+        if len(list_of_txos) > 1:
+            raise ValueError(
+                "Found more than one txout for this chat bot-initiated transaction."
+            )
+
+        return list_of_txos[0]["txo_id_hex"]
+
+    def send_payment_receipt(self, source, tx_proposal):
+        receiver_receipt = self.create_receiver_receipt(tx_proposal)
+        receiver_receipt = mc.utility.full_service_receipt_to_b64_receipt(
+            receiver_receipt
+        )
+        resp = self.signal.send_payment_receipt(source, receiver_receipt, "Refund")
+        print("Send receipt", receiver_receipt, "to", source, ":", resp)
+
+    def create_receiver_receipt(self, tx_proposal):
+        receiver_receipts = self.mcc.create_receiver_receipts(tx_proposal)
+        # I'm assuming there will only be one receiver receipt (not including change tx out).
+        if len(receiver_receipts) > 1:
+            raise ValueError(
+                "Found more than one txout for this chat bot-initiated transaction."
+            )
+        return receiver_receipts[0]
+
+    def get_unspent_pmob(self):
+        account_amount_response = self.mcc.get_balance_for_account(self.account_id)
+        unspent_pmob = int(account_amount_response["unspent_pmob"])
+        return unspent_pmob
+
+    def get_minimum_fee_pmob(self):
+        return self.minimum_fee_pmob
+
+    def handle_item_payment(self, source, customer, amount_paid_mob, drop_session):
```

Comment:
Probably out of scope given the time constraints, but we really should only ever represent values in `pmob` and only convert to `MOB` for display purposes.